frontend/src/components/Forms/variances/AddVarianceForm.js

```diff
@@ -0,0 +1,145 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { Field, reduxForm, change } from "redux-form";
+import { remove } from "lodash";
+import { Form, Button, Col, Row, Popconfirm } from "antd";
+import * as FORM from "@/constants/forms";
+import { renderConfig } from "@/components/common/config";
+import { required, dateNotInFuture, maxLength } from "@/utils/Validate";
+import { resetForm } from "@/utils/helpers";
+import VarianceFileUpload from "./VarianceFileUpload";
+import CustomPropTypes from "@/customPropTypes";
+
+const propTypes = {
+  handleSubmit: PropTypes.func.isRequired,
+  onSubmit: PropTypes.func.isRequired,
+  closeModal: PropTypes.func.isRequired,
+  submitting: PropTypes.bool.isRequired,
+  mineGuid: PropTypes.string.isRequired,
+  complianceCodes: PropTypes.arrayOf(CustomPropTypes.dropdownListItem).isRequired,
+};
+
+export class AddVarianceForm extends Component {
+  state = {
+    uploadedFiles: [],
+    documentNameGuidMap: {},
+  };
+
+  onFileLoad = (documentName, document_manager_guid) => {
+    this.state.uploadedFiles.push({ documentName, document_manager_guid });
+    this.setState(({ documentNameGuidMap }) => ({
+      documentNameGuidMap: {
+        [document_manager_guid]: documentName,
+        ...documentNameGuidMap,
+      },
+    }));
+    change("uploadedFiles", this.state.uploadedFiles);
+  };
+
+  onRemoveFile = (fileItem) => {
+    remove(this.state.uploadedFiles, { document_manager_guid: fileItem.serverId });
+    change("uploadedFiles", this.state.uploadedFiles);
+  };
+
+  render() {
+    return (
+      <Form
+        layout="vertical"
+        onSubmit={this.props.handleSubmit(this.props.onSubmit(this.state.documentNameGuidMap))}
+      >
```

Comment:
isn't `this.props.handleSubmit` and `this.props.onSubmit` the same function? 

would either of these work? 
```
onSubmit={this.props.handleSubmit(this.state.documentNameGuidMap)}
```
OR
```
onSubmit={this.props.onSubmit(this.state.documentNameGuidMap)}
```

Reply:
They're not! If I understand this correctly, it's some of the most confusing magic I've ever seen:
https://github.com/erikras/redux-form/issues/1629#issuecomment-334025455

Basically, you define your own `onSubmit`, which is your **actual** event handler. `handleSubmit` is magically provided to your component by Redux Forms. In order to use your handler (`onSubmit` in this case), you pass it as an argument to `handleSubmit`.